cla_backend/apps/cla_eventlog/tests/test_find_and_delete_old_cases.py

```diff
@@ -0,0 +1,199 @@
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 0)
+
+        self.delete_old_cases(dt)
+        self.assertEqual(Case.objects.count(), 1)
+        self.assertEqual(Complaint.objects.count(), 1)
+        self.assertEqual(Log.objects.count(), 3)
+        self.assertEqual(EODDetails.objects.count(), 1)
+
+    def test_old_case_with_recent_means_test_change(self):
+        date = _make_datetime(year=2014, month=4, day=27, hour=9)
+        case = self.create_case(date, "legalaid.eligible_case")
+
+        self.create_event_log_for_case(case, "CASE_VIEWED", _make_datetime(year=2014, month=5, day=30, hour=9))
+        self.create_event_log_for_case(case, "MT_CHANGED", _make_datetime(year=2020, month=4, day=27, hour=9))
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 0)
+
+        self.delete_old_cases(dt)
+        self.assertEqual(Case.objects.count(), 1)
+        self.assertEqual(Log.objects.count(), 2)
+        self.assertEqual(EligibilityCheck.objects.count(), 1)
+
+    def test_old_case_with_two_logs_with_same_date_of_creation(self):
+        date = _make_datetime(year=2014, month=4, day=27, hour=9)
+        case = self.create_case(date, "legalaid.eligible_case")
+
+        self.create_event_log_for_case(case, "CASE_VIEWED", _make_datetime(year=2020, month=4, day=27, hour=9))
+        self.create_event_log_for_case(case, "MT_CHANGED", _make_datetime(year=2020, month=4, day=27, hour=9))
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 0)
+
+        self.delete_old_cases(dt)
+        self.assertEqual(Case.objects.count(), 1)
+        self.assertEqual(Log.objects.count(), 2)
+        self.assertEqual(EligibilityCheck.objects.count(), 1)
+
+    def test_relatively_new_case_with_no_event_logs(self):
+        date = _make_datetime(year=2020, month=4, day=27, hour=9)
+        self.create_case(date)
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 0)
+
+        self.delete_old_cases(dt)
+        self.assertEqual(Case.objects.count(), 1)
+        self.assertEqual(Log.objects.count(), 0)
+
+    def test_old_case_with_no_event_logs(self):
+        date = _make_datetime(year=2014, month=4, day=27, hour=9)
+        self.create_case(date)
+        self.create_case(date)
+        self.create_case(date)
+
+        dt = _make_datetime(year=2021, month=1, day=1, hour=9)
+        oldCasesFound = self.find_old_cases(dt)
+        self.assertEqual(oldCasesFound.count(), 3)
+
+        # Only need to check for this once as every test case uses this method via calling self.find_old_cases
+        with self.assertNumQueries(1):
```

Comment:
Good. This is improving. However, I would put this assertion around the `self.find_old_cases(dt)` as well as the `count()` on the qs (ie the `assertNumQueries` comes before you run the command, and encapsulates both getting the qs and forcing it to evaluate).
We want to be nice and thorough, could easily miss something otherwise, like if the management command did more than return an unevaluated query. We want to know how many queries are run in total

Reply:
Cool done that

Reply:
Have you? I'm not seeing it!